code/webauthn/contracts/contracts/Account.sol

```diff
@@ -0,0 +1,345 @@
+  function validateTransaction(
+    bytes32,
+    bytes32 _suggestedSignedHash,
+    Transaction calldata _transaction
+  ) external payable override onlyBootloader returns (bytes4 magic) {
+    return _validateTransaction(_suggestedSignedHash, _transaction);
+  }
+
+  function _validateTransaction(
+    bytes32 _suggestedSignedHash,
+    Transaction calldata _transaction
+  ) internal returns (bytes4 magic) {
+    // Incrementing the nonce of the account.
+    // Note, that reserved[0] by convention is currently equal to the nonce passed in the transaction
+    SystemContractsCaller.systemCallWithPropagatedRevert(
+      uint32(gasleft()),
+      address(NONCE_HOLDER_SYSTEM_CONTRACT),
+      0,
+      abi.encodeCall(INonceHolder.incrementMinNonceIfEquals, (_transaction.nonce))
+    );
+
+    bytes32 txHash;
+    // While the suggested signed hash is usually provided, it is generally
+    // not recommended to rely on it to be present, since in the future
+    // there may be tx types with no suggested signed hash.
+    if (_suggestedSignedHash == bytes32(0)) {
+      txHash = _transaction.encodeHash();
+    } else {
+      txHash = _suggestedSignedHash;
+    }
+
+    // The fact there is are enough balance for the account
+    // should be checked explicitly to prevent user paying for fee for a
+    // transaction that wouldn't be included on Ethereum.
+    uint256 totalRequiredBalance = _transaction.totalRequiredBalance();
+    require(totalRequiredBalance <= address(this).balance, 'Not enough balance for fee + value');
+
+    // ANCHOR: _validateTransaction
+    if (isValidSignature(txHash, _transaction.signature) == EIP1271_SUCCESS_RETURN_VALUE) {
+      magic = ACCOUNT_VALIDATION_SUCCESS_MAGIC;
+    } else {
+      bool valid = validateWebAuthnSignature(_transaction.signature, txHash);
+      if (valid) {
+        magic = ACCOUNT_VALIDATION_SUCCESS_MAGIC;
+      } else {
+        magic = bytes4(0);
+      }
+    }
+    // ANCHOR_END: _validateTransaction
+  }
+
+  function executeTransaction(
+    bytes32,
+    bytes32,
+    Transaction calldata _transaction
+  ) external payable override onlyBootloader {
+    _executeTransaction(_transaction);
+  }
+
+  function _executeTransaction(Transaction calldata _transaction) internal {
+    address to = address(uint160(_transaction.to));
+    uint128 value = Utils.safeCastToU128(_transaction.value);
+    bytes memory data = _transaction.data;
+
+    if (to == address(DEPLOYER_SYSTEM_CONTRACT)) {
+      uint32 gas = Utils.safeCastToU32(gasleft());
+
+      // Note, that the deployer contract can only be called
+      // with a "systemCall" flag.
+      SystemContractsCaller.systemCallWithPropagatedRevert(gas, to, value, data);
+    } else {
+      bool success;
+      assembly {
+        success := call(gas(), to, value, add(data, 0x20), mload(data), 0, 0)
+      }
+      require(success);
+    }
+  }
+
+  function executeTransactionFromOutside(Transaction calldata _transaction) external payable {
```

Comment:
This function is better to remove at all. At least right now it is vulnerable to replay attacks from transactions initiated from `owner` address. Please note, `owner` is an EOA by itself and it can create transaction on its own account, if the nonce matches with AA account - transaction from EOA can be replayed here.